### PR TITLE
Add Private TCP Active/Active test and friends

### PR DESCRIPTION
**Pull request by aaron-lane**
## Background

This branch adds the private-tcp-active-active test, as well as necessary changes to the module to support the test.

Notable changes include:

- all Key Vault inputs are passed as objects to remove the requirement on internal data sources and dynamic URL building
- variables for certificates and secrets are clearly identified with regards to their intended usage
- removed support for deploying VM certificates through the VMSS as that approach incorrectly strips intermediate certificates from chains
- the same CA certificate secret is used for the VM and the application gateway, when defined
- the application gateway and TFE are both configured to require at least TLS v1.2

[Asana task](https://app.asana.com/0/1181500399442529/1200626076779504/f)

## How Has This Been Tested

It was tested with `/test private-tcp-active-active` (or some variation) below!

## This PR makes me feel

optional gif describing your feelings about this pr

### CODE REVIEW

File: tests/private-tcp-active-active/main.tf [hunk 1]
```diff
@@ -10,30 +10,25 @@ module "private_tcp_active_active" {
...
-  proxy_cert_secret_name = var.proxy_cert_secret_name
+  ca_certificate_secret = data.azurerm_key_vault_secret.ca_certificate
+  proxy_ip              = azurerm_linux_virtual_machine.proxy.private_ip_address
+  proxy_port            = local.proxy_port
 
   # Private Active / Active Scenario
-  vm_node_count               = 2
-  vm_sku                      = "Standard_D32a_v4"
+  vm_node_count = 2
+  # TODO: revert to "Standard_D32a_v4" when quota is increased
```
Comment (anniehedgpeth) [inline tests/private-tcp-active-active/main.tf:30]: Are you still waiting on this?

**Review by anniehedgpeth — DISMISSED** [review]
These are great changes! Thanks so much! I'm glad that we had another set of eyes this module. Nice work. :)

File: tests/private-tcp-active-active/main.tf (hunk 1 above)
Comment (aaron-lane) [reply tests/private-tcp-active-active/main.tf:30]: Oh, thanks!

Comment (aaron-lane) [reply tests/private-tcp-active-active/main.tf]: Fixed with https://github.com/hashicorp/terraform-azurerm-terraform-enterprise/pull/121/commits/6933980fe0d34749134ae98f6bb680605c7e16a8.

**Review by anniehedgpeth — APPROVED** [review]

**Pull request merged**